src/pages/SignUpPage/SignUp.jsx

```diff
@@ -1,28 +1,165 @@
-import React from 'react';
+import React, { useReducer, useState } from 'react';
 import { Link } from 'react-router-dom';
 import "./SignUp.css";
+import signUpAnimation from "./../../../src/assets/images/SVG/signup.svg"
+import axios from 'axios';
+import { useNavigate as navigate } from 'react-router';
+import { useAuth } from '../../context/AuthContext';
+
+const SignUpDetails = (state,action) =>{
+    console.log(state,action);
+    console.log(action.email,action.firstName,action.lastName)
+    if(action.email){
+        return {...state,email : action.email}
+    }
+    if(action.firstName){
+        return {...state,firstName : action.firstName}
+    }
+    if(action.lastName){
+        return {...state,lastName : action.lastName}
+    }
+    return {...state}
+}
+
```

Comment:
You can use here `switch` statement as it more readable than traditional `if` control statement